python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+    initdist_ind.add(ot.Uniform(lb_ind[i], ub_ind[i]))
+initdist_ind = ot.ComposedDistribution(initdist_ind)
+initSample_ind = initdist_ind.getSample(10)
+optalg_ind = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_ind)
+
+# %%
+# We generate the training data set
+X = dist.getSample(10)
+Y = fun(X)
+
+# And the plotting data set
+Xplt = dist.getSample(200)
+Xplt = Xplt.sort()
+Yplt = fun(Xplt)
+
+# %%
+# Initialize  and parameterize the optimization algorithm
+initSample_LV = initdist_LV.getSample(30)
+optalg_LV = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_LV)
+
+# %%
+# Create and train the Gaussian process models
+basis = ot.ConstantBasisFactory(2).build()
+algo_LV = ot.KrigingAlgorithm(X, Y, kLV, basis)
+algo_LV.setOptimizationAlgorithm(optalg_LV)
+algo_LV.setOptimizationBounds(bounds_LV)
+algo_LV.run()
+res_LV = algo_LV.getResult()
+
+algo_ind_list = []
+for z in range(2):
+    # We select the training samples corresponding to the correct combination
+    # of categorical levels
+    ind = np.where(np.all(np.array(X[:, 1]) == z, axis=1))[0]
+    Xl = X[ind][:, 0]
+    Yl = Y[ind]
+
+    # Create and train the Gaussian process models
+    basis = ot.ConstantBasisFactory(1).build()
+    algo_ind = ot.KrigingAlgorithm(Xl, Yl, k_ind, basis)
+    algo_ind.setOptimizationAlgorithm(optalg_ind)
+    algo_ind.setOptimizationBounds(bounds_ind)
+    algo_ind.run()
+    algo_ind_list.append(algo_ind.getResult())
+
+# %%
+# We plot the prediction of the mixed continuous / categorical GP,
+# as well as the one of the two separate continuous GPs
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True, figsize=(15, 10))
+for z in range(2):
+    # We select the training samples corresponding to the correct combination
+    # of categorical levels
+    ind = np.where(np.all(np.array(X[:, 1]) == z, axis=1))[0]
+    Xl = X[ind][:, 0]
+    Yl = Y[ind]
+
+    # Compute the models predictive performances on a validation data set
+    # We compute the predictions independently for each level of z
+    ind = np.where(np.all(np.array(Xplt[:, 1]) == z, axis=1))[0]
+    Xplt_ind = Xplt[ind]
+    Yplt_ind = Yplt[ind]
+
+    predMeanLV = res_LV.getConditionalMean(Xplt_ind)
+    predMeanInd = algo_ind_list[z].getConditionalMean(Xplt_ind[:, 0])
+    predSTDLV = np.sqrt(res_LV.getConditionalMarginalVariance(Xplt_ind))
+    predSTDInd = np.sqrt(
+        algo_ind_list[z].getConditionalMarginalVariance(Xplt_ind[:, 0])
+    )
+
+    (trainingData,) = ax1.plot(Xl[:, 0], Yl, "r*")
+    (trueFunction,) = ax1.plot(Xplt_ind[:, 0], Yplt_ind, "k--")
+    (prediction,) = ax1.plot(Xplt_ind[:, 0], predMeanLV, "b-")
+    stdPred = ax1.fill_between(
+        Xplt_ind[:, 0].asPoint(),
+        (predMeanLV - predSTDLV).asPoint(),
+        (predMeanLV + predSTDLV).asPoint(),
+        alpha=0.5,
+        color="blue",
+    )
+    ax2.plot(Xl[:, 0], Yl, "r*")
+    ax2.plot(Xplt_ind[:, 0], Yplt_ind, "k--")
+    ax2.plot(Xplt_ind[:, 0], predMeanInd, "b-")
+    ax2.fill_between(
```

Comment:
This is shown [here](http://openturns.github.io/openturns/latest/auto_graphs/plot_graphs_fill_area.html) in OT.